src/main/java/com/oneops/api/resource/Transition.java

```diff
@@ -1889,6 +1988,96 @@ public CiResource updateRelay(String environmentName, String relayName, String s
 		throw new OneOpsClientAPIException(msg);
 	}
 
+	/**
+	 * 1. The first Deployment API call generates a deployment plan, starts the deployment and returns
+	 * the deployment plan information.
+	 * 2. We need a second API call to check if the deployment started successfully or if it failed due
+	 * to capacity issues (if deploy is called without committing), or any other errors.
+	 * If the deployment has started successfully, sometimes we do get `Active deployment in progress` error
+	 * with the second API call, which can be ignored.
+	 * @param environmentName
+	 * @param request
+	 * @param jsonObject
+	 * @throws OneOpsClientAPIException
+	 */
+
+	private void deployAndCheckStatus(String environmentName, RequestSpecification request,
+											 JSONObject jsonObject) throws OneOpsClientAPIException {
+		// Start a new deployment
+		Response response = doDeploy(environmentName, transitionEnvUri, request, jsonObject);
+		checkForErrors(environmentName, response, IConstants.EXISTING_DEPLOYMENT_MESSAGE);
+
+		// Wait for 5s before checking if the deployment started successfully
+		try {
+			Thread.sleep(5000);
+		} catch(InterruptedException ie) {
+			LOG.error("Error trying to delay: {}", ie.getMessage());
+		}
+
+		// Check for capacity issues and if deployment has started or failed due to capacity error.
+		response = doDeploy(environmentName, transitionEnvUri, request, jsonObject);
```

Comment:
Why is this second api call required ? If its just to check the outcome of the first api call,  we should just make an api call to fetch the latest deployment for the environment `/<org_name>/assemblies/<assembly_name>/transition/environments/<env_name>/deployments/latest` and see if the status is active. Additionally, to be absolutely sure that the latest deployment is indeed the one initiated from concord (and not something that was triggered in between directly in OneOps, or from a previous attempt from concord itself) , you could pass in a random uuid (kind of like a correlation id) while starting the deployment and the compare the comments while fetching the latest deployment as mentioned above.

Reply:
I went through the implementation once again , and have a better way to check whether deployment initiation was successful after the first api call : 

 Fetch the environment itself (using `/<org_name>/assemblies/<assembly_name>/transition/environments/<env_name>`) and check the `ciState` and `comments`. 

- If deployment initiation is still going on, `ciState` would be locked. In this case, you need to retry the check until `ciState` changes to `default` as explained below.
- For successful deployment initiation,  `ciState` would be `default` and `comments` would be something like this `SUCCESS: Generation time taken: 3.537 seconds. bomGenerationInfo={\"mode\":\"persistent\",\"manifestCommit\":false,\"rfcCiCount\":2,\"createdBy\":\"a0p01pt\",\"autoDeploy\":true,\"releaseId\":1748,\"rfcRelationCount\":2}`. 
- In case deployment initiation failed,  `ciState` would be still `default` , but `comments` would be something like this `ERROR:BOM:There is an active deployment 1741 in this environment, you need to cancel or retry it`

Reply:
I have made changes to the `deploy` method and this is the logic it follows now:
1. The first Deployment API call generates a deployment plan, starts the deployment and returns the environment information.
2. Check the environment API till CI state changes from "locked" to "default". When CI state becomes "default", the deployment has been initiated/failed.
3. Check the message of the latest deployment and if that matches the comments provided to the method call.
4. If the comments match return the latest deployment object
5. If it does not match, that means the deployment was not successfully initiated. 
    a. Check for `comments` of the environment to see if there was any ERROR. Throw `OneOpsClientAPIException` with the message.
    b. If there are no errors in `comments` of the environment, throw `OneOpsClientAPIException` with `DEFAULT_ERROR_MESSAGE`.